compiler/x/runtime/X86Runtime.hpp

```diff
@@ -71,4 +74,30 @@ inline bool jitGetCPUID(TR_X86CPUIDBuffer* pBuffer)
       }
    }
 
+inline bool InterlockedCompareAndSwap(volatile uint32_t* ptr, uint32_t old_val, uint32_t new_val)
+   {
+#ifdef WINDOWS
+   return old_val == (uint32_t)_InterlockedCompareExchange((volatile long*)ptr, (long)new_val, (long)old_val);
+#else
+   return __sync_bool_compare_and_swap(ptr, old_val, new_val);
+#endif
+   }
+
+inline bool InterlockedCompareAndSwap(volatile uint16_t* ptr, uint16_t old_val, uint16_t new_val)
+   {
+#ifdef WINDOWS
+   return old_val == (uint16_t)_InterlockedCompareExchange16((volatile short*)ptr, (short)new_val, (short)old_val);
+#else
+   return __sync_bool_compare_and_swap(ptr, old_val, new_val);
+#endif
+   }
+
+inline void X86PatchingFence16(void* addr)
```

Comment:
Why did you call this `X86PatchingFence16` as opposed to simply `patchingFence16`?  It just feels a little awkward.

Reply:
I actually like the name `patchingFence16` more. It has been updated.